### ISSUE TITLE
Idea: Transform PointNet into a 6DoF localization algorithm

### ISSUE THREAD

**Issue by DanielCoelho112** [DanielCoelho112/localization_end_to_end#10]
Hi @miguelriemoliveira and @pmdjdias,

I did some research about the integration of point clouds in neural networks and found something interesting. There are 4 ways of inputting this data:

- a set of images;
- a voxel grid;
- a graph;
- an unordered set of points.

The latest and most promising approach is "an unordered set of points" and it was proposed in 2017 in this [paper](https://arxiv.org/abs/1612.00593). This approach is called PointNet. Later that year, the same authors proposed an upgrade of the method ([PointNet++](https://arxiv.org/abs/1706.02413)).  

The applications targeted by these approaches include object classification, part segmentation, and semantic segmentation. I've looked over the architectures of these approaches and I think we can adapt them to tackle the localization problem. In fact, the only modifications needed are the removal of two feature transformations and the removal of the SoftMax activation function. Given the popularity of these approaches in the point cloud community, there are a lot of implementations in pytorch that we could explore.

So I think we could explore this approach and build an architecture that receives point clouds as input and estimates the 6DoF. If we follow this path, we are postponing our multimodal approach and I think it is the safest approach. Given that we are just starting out in this area, I think that starting with something that no one has done so far can be really difficult.

I want to hear your opinions on this. If you want we can schedule a meeting to talk about this in-depth.
